### PR TITLE
add some unit tests in staging/src/k8s.io/apiserver/pkg/server/options

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
add some unit tests in staging/src/k8s.io/apiserver/pkg/server/options
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilflag "k8s.io/apiserver/pkg/util/flag"
+)
+
+type fakeGroupRegisty struct{}
+
+func (f fakeGroupRegisty) IsGroupRegistered(group string) bool {
+	return group == "apiregistration.k8s.io"
+}
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go]: please add empty line below this `}`

File: staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"strings"
+	"testing"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilflag "k8s.io/apiserver/pkg/util/flag"
+)
+
+type fakeGroupRegisty struct{}
+
+func (f fakeGroupRegisty) IsGroupRegistered(group string) bool {
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go]: hm, i can't find usage of this function anywhere.
is something missing?

also should be lowercase if not used outside of the package.

File: staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			},
+		},
+	}
+	testGroupRegisty := fakeGroupRegisty{}
+
+	for _, testcase := range testCases {
+		t.Run(testcase.name, func(t *testing.T) {
+			errs := testcase.testOptions.Validate(testGroupRegisty)
+			if len(testcase.expectErr) != 0 && !strings.Contains(utilerrors.NewAggregate(errs).Error(), testcase.expectErr) {
+				t.Errorf("Got err: %v, Expected err: %s", errs, testcase.expectErr)
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go]: errors strings should start with lowercase:
`"got err: %v, expected err: %s"`

should this be `t.Fatalf()` ?

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd_test.go [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+				EtcdServersOverrides:    []string{"/events#http://127.0.0.1:4002"},
+			},
+		},
+	}
+
+	for _, testcase := range testCases {
+		t.Run(testcase.name, func(t *testing.T) {
+			errs := testcase.testOptions.Validate()
+			if len(testcase.expectErr) != 0 && !strings.Contains(utilerrors.NewAggregate(errs).Error(), testcase.expectErr) {
+				t.Errorf("Got err: %v, Expected err: %s", errs, testcase.expectErr)
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd_test.go]: should this be `t.Fatalf()`?

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd_test.go [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+	for _, testcase := range testCases {
+		t.Run(testcase.name, func(t *testing.T) {
+			result, err := ParseWatchCacheSizes(testcase.cacheSizes)
+			if len(testcase.expectErr) != 0 && !strings.Contains(err.Error(), testcase.expectErr) {
+				t.Errorf("Got err: %v, Expected err: %s", err, testcase.expectErr)
+			}
+			if len(testcase.expectErr) == 0 {
+				if err != nil {
+					t.Errorf("Got err: %v, Expected err nil", err)
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd_test.go]: if this is `t.Fatalf()` instead the bellow `} else {` can be removed.

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd_test.go [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+		},
+	}
+
+	for _, testcase := range testCases {
+		t.Run(testcase.name, func(t *testing.T) {
+			result, err := ParseWatchCacheSizes(testcase.cacheSizes)
+			if len(testcase.expectErr) != 0 && !strings.Contains(err.Error(), testcase.expectErr) {
+				t.Errorf("Got err: %v, Expected err: %s", err, testcase.expectErr)
+			}
+			if len(testcase.expectErr) == 0 {
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd_test.go]:
```
if len(testcase.expectErr) != 0 {
	continue
}
```

File: staging/src/k8s.io/apiserver/pkg/server/options/server_run_options_test.go [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+				TargetRAMMB:                 65536,
+			},
+		},
+	}
+
+	for _, testcase := range testCases {
+		t.Run(testcase.name, func(t *testing.T) {
+			errs := testcase.testOptions.Validate()
+			if len(testcase.expectErr) != 0 && !strings.Contains(utilerrors.NewAggregate(errs).Error(), testcase.expectErr) {
+				t.Errorf("Got err: %v, Expected err: %s", errs, testcase.expectErr)
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/server/options/server_run_options_test.go]: `t.Fatalf()`?

**Comment by neolit123** [comment]
thanks for this PR with unit tests.
added some minor comments.

File: staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go (hunk 2 above)
Comment (hanxiaoshuai) [reply staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go]: It is a GroupRegisty interface in https://github.com/kubernetes/kubernetes/blob/09cdfc52872fbf4a6a3bfcad54baf216c1238261/staging/src/k8s.io/apiserver/pkg/server/options/api_enablement.go#L107-L110

File: staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go (hunk 3 above)
Comment (hanxiaoshuai) [reply staging/src/k8s.io/apiserver/pkg/server/options/api_enablement_test.go]: I think we should use t.Errorf here.  t.Fatalf() will stop other cases run in this UT

**Pull request merged**